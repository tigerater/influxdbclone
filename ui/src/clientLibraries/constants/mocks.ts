export interface ClientLibrary {
  id: string
  name: string
  description?: string
}

export const mockClientLibraries: ClientLibrary[] = [
  {
    id: 'client-lib-js',
    name: 'Javascript',
    description: 'Write to InfuxDB using Javascript',
  },
  {
    id: 'client-lib-node',
    name: 'Node',
    description: 'Write to InfuxDB using Node',
  },
  {
    id: 'client-lib-go',
    name: 'GO',
    description: 'Write to InfuxDB using GO',
  },
  {
    id: 'client-lib-python',
    name: 'Python',
    description: 'Write to InfuxDB using Python',
  },
]
