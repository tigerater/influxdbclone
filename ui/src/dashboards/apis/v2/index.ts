// Libraries
import _ from 'lodash'

// Utils
import {addLabelDefaults} from 'src/shared/utils/labels'
import {incrementCloneName} from 'src/utils/naming'

// Types

import {Cell, NewCell, Dashboard, View} from 'src/types/v2'
import {Label} from '@influxdata/influx'

import {Cell as CellAPI} from '@influxdata/influx'
import {client} from 'src/utils/api'

export const addDashboardIDToCells = (
  cells: CellAPI[],
  dashboardID: string
): Cell[] => {
  return cells.map(c => {
    return {...c, dashboardID}
  })
}

// TODO(desa): what to do about getting dashboards from another v2 source
export const getDashboards = async (): Promise<Dashboard[]> => {
  const dashboards = await client.dashboards.getAll()

  return dashboards.map(d => ({
    ...d,
    labels: d.labels.map(addLabelDefaults),
    cells: addDashboardIDToCells(d.cells, d.id),
  }))
}

export const getDashboard = async (id: string): Promise<Dashboard> => {
  const dashboard = await client.dashboards.get(id)

  return {
    ...dashboard,
    labels: dashboard.labels.map(addLabelDefaults),
    cells: addDashboardIDToCells(dashboard.cells, dashboard.id),
  }
}

export const createDashboard = async (
  props: Partial<Dashboard>
): Promise<Dashboard> => {
  const dashboard = await client.dashboards.create(props)

  return {
    ...dashboard,
    labels: dashboard.labels.map(addLabelDefaults),
    cells: addDashboardIDToCells(dashboard.cells, dashboard.id),
  }
}

export const deleteDashboard = async (dashboard: Dashboard): Promise<void> => {
  await client.dashboards.delete(dashboard.id)
}

export const updateDashboard = async (
  dashboard: Dashboard
): Promise<Dashboard> => {
  const updated = await client.dashboards.update(dashboard.id, dashboard)

  return {
    ...updated,
    labels: updated.labels.map(addLabelDefaults),
    cells: addDashboardIDToCells(updated.cells, updated.id),
  }
}

export const addCell = async (
  dashboardID: string,
  cell: NewCell
): Promise<Cell> => {
  const result = await client.dashboards.createCell(dashboardID, cell)

  const cellWithID = {...result, dashboardID}

  return cellWithID
}

export const updateCells = async (
  id: string,
  cells: Cell[]
): Promise<Cell[]> => {
  const result = await client.dashboards.updateAllCells(id, cells)

  return addDashboardIDToCells(result, id)
}

export const deleteCell = async (
  dashboardID: string,
  cell: Cell
): Promise<void> => {
  await client.dashboards.deleteCell(dashboardID, cell.id)
}

export const addDashboardLabels = async (
  dashboardID: string,
  labels: Label[]
): Promise<Label[]> => {
  const addedLabels = await Promise.all(
    labels.map(async label => {
      return client.dashboards.createLabel(dashboardID, label.id)
    })
  )

  return addedLabels
}

export const removeDashboardLabels = async (
  dashboardID: string,
  labels: Label[]
): Promise<void> => {
  await Promise.all(
    labels.map(label => {
      return client.dashboards.deleteLabel(dashboardID, label.id)
    })
  )
}

export const readView = async (
  dashboardID: string,
  cellID: string
): Promise<View> => {
  const data = await client.dashboards.getView(dashboardID, cellID)

  const view: View = {...data, dashboardID, cellID}

  return view
}

export const updateView = async (
  dashboardID: string,
  cellID: string,
  view: Partial<View>
): Promise<View> => {
  const data = await client.dashboards.updateView(dashboardID, cellID, view)

  const viewWithIDs: View = {...data, dashboardID, cellID}

  return viewWithIDs
}

export const cloneDashboard = async (
  dashboardToClone: Dashboard,
  dashboards: Dashboard[]
) => {
  const allDashboardNames = dashboards.map(d => d.name)

  const clonedName = incrementCloneName(
    allDashboardNames,
    dashboardToClone.name
  )

  if (dashboardToClone.id) {
    return client.dashboards.clone(dashboardToClone.id, clonedName)
  }
}
