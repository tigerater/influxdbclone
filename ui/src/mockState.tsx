import React from 'react'
import {Provider} from 'react-redux'

import {render} from 'react-testing-library'
import configureStore from 'src/store/configureStore'
import {createMemoryHistory} from 'history'

const localState = {
  app: {
    ephemeral: {
      inPresentationMode: false,
    },
    persisted: {autoRefresh: 0, showTemplateControlBar: false},
  },
  VERSION: '2.0.0',
  ranges: [
    {
      dashboardID: '0349ecda531ea000',
      seconds: 900,
      lower: 'now() - 15m',
      upper: null,
      label: 'Past 15m',
      duration: '15m',
    },
  ],
}

const history = createMemoryHistory({entries: ['/']})

export function renderWithRedux(ui, initialState = s => s) {
  const seedStore = configureStore(localState, history)
  const seedState = seedStore.getState()
  const store = configureStore(initialState(seedState), history)

  const provider = <Provider store={store}>{ui}</Provider>

  return {
    ...render(provider),
    store,
  }
}
