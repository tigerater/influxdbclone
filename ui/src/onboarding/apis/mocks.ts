import {
  getTelegrafConfigsResponse,
  createTelegrafConfigResponse,
  setSetupParamsResponse,
} from 'mocks/dummyData'

const telegrafsGet = jest.fn(() =>
  Promise.resolve(getTelegrafConfigsResponse.data)
)
const telegrafsPost = jest.fn(() =>
  Promise.resolve(createTelegrafConfigResponse.data)
)
const telegrafsTelegrafIDPut = jest.fn(() =>
  Promise.resolve(createTelegrafConfigResponse.data)
)
const setupPost = jest.fn(() => Promise.resolve(setSetupParamsResponse))
const setupGet = jest.fn(() => Promise.resolve({data: {allowed: true}}))

export const telegrafsAPI = {
  telegrafsGet,
  telegrafsPost,
  telegrafsTelegrafIDPut,
}

export const client = {
  telegrafConfigs: {
    getAll: telegrafsGet,
    getAllByOrg: telegrafsGet,
    create: telegrafsPost,
  },
}

export const setupAPI = {
  setupPost,
  setupGet,
}
