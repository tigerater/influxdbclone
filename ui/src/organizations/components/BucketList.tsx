// Libraries
import React, {PureComponent} from 'react'
import {withRouter, WithRouterProps} from 'react-router'
import {connect} from 'react-redux'
import _ from 'lodash'

// Components
import UpdateBucketOverlay from 'src/organizations/components/UpdateBucketOverlay'
import BucketRow, {PrettyBucket} from 'src/organizations/components/BucketRow'
import {OverlayTechnology, IndexList} from 'src/clockface'
import DataLoaderSwitcher from 'src/dataLoaders/components/DataLoaderSwitcher'

// Actions
import {setBucketInfo} from 'src/dataLoaders/actions/steps'

// Types
import {OverlayState} from 'src/types/v2'
import {Substep, DataLoaderStep, DataLoaderType} from 'src/types/v2/dataLoaders'
import {setDataLoadersType} from 'src/dataLoaders/actions/dataLoaders'
import {AppState} from 'src/types/v2'

interface OwnProps {
  buckets: PrettyBucket[]
  emptyState: JSX.Element
  onUpdateBucket: (b: PrettyBucket) => Promise<void>
  onDeleteBucket: (b: PrettyBucket) => Promise<void>
}

interface DispatchProps {
  onSetBucketInfo: typeof setBucketInfo
  onSetDataLoadersType: typeof setDataLoadersType
}

interface StateProps {
  dataLoaderType: DataLoaderType
}

type Props = OwnProps & StateProps & DispatchProps

interface State {
  bucketID: string
  bucketOverlayState: OverlayState
  dataLoadersOverlayState: OverlayState
}

class BucketList extends PureComponent<Props & WithRouterProps, State> {
  constructor(props) {
    super(props)

    const openDataLoaderOverlay = _.get(
      this,
      'props.location.query.openDataLoaderOverlay',
      false
    )
    const firstBucketID = _.get(this, 'props.buckets.0.id', null)
    const bucketID = openDataLoaderOverlay ? firstBucketID : null

    this.state = {
      bucketID,
      bucketOverlayState: OverlayState.Closed,
      dataLoadersOverlayState: openDataLoaderOverlay
        ? OverlayState.Open
        : OverlayState.Closed,
    }
  }

  public render() {
    const {dataLoaderType, buckets, emptyState, onDeleteBucket} = this.props

    return (
      <>
        <IndexList>
          <IndexList.Header>
            <IndexList.HeaderCell columnName="Name" width="40%" />
            <IndexList.HeaderCell columnName="Retention" width="40%" />
            <IndexList.HeaderCell columnName="" width="20%" />
          </IndexList.Header>
          <IndexList.Body columnCount={3} emptyState={emptyState}>
            {buckets.map(bucket => (
              <BucketRow
                key={bucket.id}
                bucket={bucket}
                onEditBucket={this.handleStartEdit}
                onDeleteBucket={onDeleteBucket}
                onAddData={this.handleStartAddData}
                onUpdateBucket={this.handleUpdateBucket}
              />
            ))}
          </IndexList.Body>
        </IndexList>
        <OverlayTechnology visible={this.isBucketOverlayVisible}>
          <UpdateBucketOverlay
            bucket={this.bucket}
            onCloseModal={this.handleCloseModal}
            onUpdateBucket={this.handleUpdateBucket}
          />
        </OverlayTechnology>
        <DataLoaderSwitcher
          type={dataLoaderType}
          visible={this.isDataLoadersWizardVisible}
          onCompleteSetup={this.handleDismissDataLoaders}
          buckets={buckets}
          {...this.startingValues}
        />
      </>
    )
  }

  private get bucket(): PrettyBucket {
    return this.props.buckets.find(b => b.id === this.state.bucketID)
  }

  private get startingValues(): {
    startingType: DataLoaderType
    startingStep: number
    startingSubstep?: Substep
  } {
    const {dataLoaderType} = this.props

    switch (dataLoaderType) {
      case DataLoaderType.Scraping:
        return {
          startingType: DataLoaderType.Scraping,
          startingStep: DataLoaderStep.Configure,
        }
      case DataLoaderType.LineProtocol:
        return {
          startingType: DataLoaderType.LineProtocol,
          startingStep: DataLoaderStep.Configure,
        }
    }
  }

  private handleCloseModal = () => {
    this.setState({bucketOverlayState: OverlayState.Closed})
  }

  private handleStartEdit = (bucket: PrettyBucket) => {
    this.setState({bucketID: bucket.id, bucketOverlayState: OverlayState.Open})
  }

  private handleStartAddData = (
    bucket: PrettyBucket,
    dataLoaderType: DataLoaderType
  ) => {
    this.props.onSetBucketInfo(
      bucket.organization,
      bucket.organizationID,
      bucket.name,
      bucket.id
    )

    this.props.onSetDataLoadersType(dataLoaderType)

    this.setState({
      bucketID: bucket.id,
      dataLoadersOverlayState: OverlayState.Open,
    })
  }

  private handleDismissDataLoaders = () => {
    this.setState({
      bucketID: '',
      dataLoadersOverlayState: OverlayState.Closed,
    })
  }

  private get isDataLoadersWizardVisible(): boolean {
    const {bucketID, dataLoadersOverlayState} = this.state
    return !!bucketID && dataLoadersOverlayState === OverlayState.Open
  }

  private get isBucketOverlayVisible(): boolean {
    const {bucketID, bucketOverlayState} = this.state
    return !!bucketID && bucketOverlayState === OverlayState.Open
  }

  private handleUpdateBucket = async (updatedBucket: PrettyBucket) => {
    await this.props.onUpdateBucket(updatedBucket)
    this.setState({bucketOverlayState: OverlayState.Closed})
  }
}

const mstp = ({
  dataLoading: {
    dataLoaders: {type},
  },
}: AppState): StateProps => ({
  dataLoaderType: type,
})

const mdtp: DispatchProps = {
  onSetBucketInfo: setBucketInfo,
  onSetDataLoadersType: setDataLoadersType,
}

export default connect<StateProps, DispatchProps, OwnProps>(
  mstp,
  mdtp
)(withRouter<Props>(BucketList))
