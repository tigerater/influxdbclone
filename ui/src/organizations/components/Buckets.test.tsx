// Libraries
import React from 'react'
import {renderWithRedux} from 'src/mockState'

// Components
import BucketList from 'src/organizations/components/BucketList'

// Constants
import {buckets} from 'mocks/dummyData'
import {DataLoaderType} from 'src/types/v2/dataLoaders'
import {withRouterProps} from 'mocks/dummyData'

const setup = (override?) => {
  const props = {
    ...withRouterProps,
    buckets,
    emptyState: <></>,
    onUpdateBucket: jest.fn(),
    onDeleteBucket: jest.fn(),
    onSetBucketInfo: jest.fn(),
    onSetDataLoadersType: jest.fn(),
    dataLoaderType: DataLoaderType.Streaming,
    ...override,
  }

  const wrapper = renderWithRedux(<BucketList {...props} />)

  return {wrapper}
}

describe('BucketList', () => {
  describe('rendering', () => {
    it('renders', () => {
      const {wrapper} = setup()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
