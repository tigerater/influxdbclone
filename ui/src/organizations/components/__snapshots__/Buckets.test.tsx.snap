// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BucketList rendering renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="index-list"
      >
        <thead
          class="index-list--header"
        >
          <tr>
            <th
              class="index-list--header-cell index-list--align-left"
              style="width: 40%;"
            >
              Name
            </th>
            <th
              class="index-list--header-cell index-list--align-left"
              style="width: 40%;"
            >
              Retention
            </th>
            <th
              class="index-list--header-cell index-list--align-left"
              style="width: 20%;"
            >
              
            </th>
          </tr>
        </thead>
        <tbody
          class="index-list--body"
        >
          <tr
            class="index-list--row"
          >
            <td
              class="index-list--row-cell index-list--align-left"
            >
              <div
                class="index-list--cell"
              >
                <div
                  class="editable-name"
                >
                  <a
                    href="#"
                  >
                    <span>
                      newbuck
                    </span>
                  </a>
                  <div
                    class="editable-name--toggle"
                  >
                    <span
                      class="icon pencil"
                    />
                  </div>
                </div>
              </div>
            </td>
            <td
              class="index-list--row-cell index-list--align-left"
            >
              <div
                class="index-list--cell"
              />
            </td>
            <td
              class="index-list--row-cell index-list--show-hover index-list--align-right"
            >
              <div
                class="index-list--cell"
              >
                <div
                  class="confirmation-button"
                >
                  <button
                    class="button button-xs button-danger"
                    tabindex="0"
                    title="Delete"
                    type="button"
                  >
                    Delete
                  </button>
                  <div
                    class="confirmation-button--tooltip"
                  >
                    <div
                      class="confirmation-button--tooltip-body"
                      data-test="confirmation-button--click-target"
                    >
                      Confirm
                    </div>
                  </div>
                </div>
              </div>
            </td>
            <td
              class="index-list--row-cell index-list--align-right"
            >
              <div
                class="index-list--cell"
              >
                <div
                  class="context-menu"
                >
                  <div
                    class="context-menu--container"
                  >
                    <button
                      class="button button-xs button-primary context-menu--toggle context-menu--primary"
                      tabindex="0"
                      title="Add Data"
                      type="button"
                    >
                      Add Data
                    </button>
                    <div
                      class="context-menu--list-container"
                    >
                      <div
                        class="context-menu--list context-menu--primary"
                      >
                        <button
                          class="context-menu--item"
                        >
                          Configure Telegraf Agent
                          <div
                            class="contex-menu--item-description"
                          >
                            Configure a Telegraf agent to push data into your bucket.
                          </div>
                        </button>
                        <button
                          class="context-menu--item"
                        >
                          Line Protocol
                          <div
                            class="contex-menu--item-description"
                          >
                            Quickly load an existing line protocol file.
                          </div>
                        </button>
                        <button
                          class="context-menu--item"
                        >
                          Scrape Metrics
                          <div
                            class="contex-menu--item-description"
                          >
                            Add a scrape target to pull data into your bucket.
                          </div>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </td>
          </tr>
          <tr
            class="index-list--row"
          >
            <td
              class="index-list--row-cell index-list--align-left"
            >
              <div
                class="index-list--cell"
              >
                <div
                  class="editable-name"
                >
                  <a
                    href="#"
                  >
                    <span>
                      newbuck1
                    </span>
                  </a>
                  <div
                    class="editable-name--toggle"
                  >
                    <span
                      class="icon pencil"
                    />
                  </div>
                </div>
              </div>
            </td>
            <td
              class="index-list--row-cell index-list--align-left"
            >
              <div
                class="index-list--cell"
              />
            </td>
            <td
              class="index-list--row-cell index-list--show-hover index-list--align-right"
            >
              <div
                class="index-list--cell"
              >
                <div
                  class="confirmation-button"
                >
                  <button
                    class="button button-xs button-danger"
                    tabindex="0"
                    title="Delete"
                    type="button"
                  >
                    Delete
                  </button>
                  <div
                    class="confirmation-button--tooltip"
                  >
                    <div
                      class="confirmation-button--tooltip-body"
                      data-test="confirmation-button--click-target"
                    >
                      Confirm
                    </div>
                  </div>
                </div>
              </div>
            </td>
            <td
              class="index-list--row-cell index-list--align-right"
            >
              <div
                class="index-list--cell"
              >
                <div
                  class="context-menu"
                >
                  <div
                    class="context-menu--container"
                  >
                    <button
                      class="button button-xs button-primary context-menu--toggle context-menu--primary"
                      tabindex="0"
                      title="Add Data"
                      type="button"
                    >
                      Add Data
                    </button>
                    <div
                      class="context-menu--list-container"
                    >
                      <div
                        class="context-menu--list context-menu--primary"
                      >
                        <button
                          class="context-menu--item"
                        >
                          Configure Telegraf Agent
                          <div
                            class="contex-menu--item-description"
                          >
                            Configure a Telegraf agent to push data into your bucket.
                          </div>
                        </button>
                        <button
                          class="context-menu--item"
                        >
                          Line Protocol
                          <div
                            class="contex-menu--item-description"
                          >
                            Quickly load an existing line protocol file.
                          </div>
                        </button>
                        <button
                          class="context-menu--item"
                        >
                          Scrape Metrics
                          <div
                            class="contex-menu--item-description"
                          >
                            Add a scrape target to pull data into your bucket.
                          </div>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
      <div
        class="overlay-tech"
      >
        <div
          class="overlay--dialog"
          data-test="overlay-children"
        />
        <div
          class="overlay--mask"
        />
      </div>
      <div
        class="overlay-tech"
      >
        <div
          class="overlay--dialog"
          data-test="overlay-children"
        />
        <div
          class="overlay--mask"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <table
      class="index-list"
    >
      <thead
        class="index-list--header"
      >
        <tr>
          <th
            class="index-list--header-cell index-list--align-left"
            style="width: 40%;"
          >
            Name
          </th>
          <th
            class="index-list--header-cell index-list--align-left"
            style="width: 40%;"
          >
            Retention
          </th>
          <th
            class="index-list--header-cell index-list--align-left"
            style="width: 20%;"
          >
            
          </th>
        </tr>
      </thead>
      <tbody
        class="index-list--body"
      >
        <tr
          class="index-list--row"
        >
          <td
            class="index-list--row-cell index-list--align-left"
          >
            <div
              class="index-list--cell"
            >
              <div
                class="editable-name"
              >
                <a
                  href="#"
                >
                  <span>
                    newbuck
                  </span>
                </a>
                <div
                  class="editable-name--toggle"
                >
                  <span
                    class="icon pencil"
                  />
                </div>
              </div>
            </div>
          </td>
          <td
            class="index-list--row-cell index-list--align-left"
          >
            <div
              class="index-list--cell"
            />
          </td>
          <td
            class="index-list--row-cell index-list--show-hover index-list--align-right"
          >
            <div
              class="index-list--cell"
            >
              <div
                class="confirmation-button"
              >
                <button
                  class="button button-xs button-danger"
                  tabindex="0"
                  title="Delete"
                  type="button"
                >
                  Delete
                </button>
                <div
                  class="confirmation-button--tooltip"
                >
                  <div
                    class="confirmation-button--tooltip-body"
                    data-test="confirmation-button--click-target"
                  >
                    Confirm
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            class="index-list--row-cell index-list--align-right"
          >
            <div
              class="index-list--cell"
            >
              <div
                class="context-menu"
              >
                <div
                  class="context-menu--container"
                >
                  <button
                    class="button button-xs button-primary context-menu--toggle context-menu--primary"
                    tabindex="0"
                    title="Add Data"
                    type="button"
                  >
                    Add Data
                  </button>
                  <div
                    class="context-menu--list-container"
                  >
                    <div
                      class="context-menu--list context-menu--primary"
                    >
                      <button
                        class="context-menu--item"
                      >
                        Configure Telegraf Agent
                        <div
                          class="contex-menu--item-description"
                        >
                          Configure a Telegraf agent to push data into your bucket.
                        </div>
                      </button>
                      <button
                        class="context-menu--item"
                      >
                        Line Protocol
                        <div
                          class="contex-menu--item-description"
                        >
                          Quickly load an existing line protocol file.
                        </div>
                      </button>
                      <button
                        class="context-menu--item"
                      >
                        Scrape Metrics
                        <div
                          class="contex-menu--item-description"
                        >
                          Add a scrape target to pull data into your bucket.
                        </div>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </td>
        </tr>
        <tr
          class="index-list--row"
        >
          <td
            class="index-list--row-cell index-list--align-left"
          >
            <div
              class="index-list--cell"
            >
              <div
                class="editable-name"
              >
                <a
                  href="#"
                >
                  <span>
                    newbuck1
                  </span>
                </a>
                <div
                  class="editable-name--toggle"
                >
                  <span
                    class="icon pencil"
                  />
                </div>
              </div>
            </div>
          </td>
          <td
            class="index-list--row-cell index-list--align-left"
          >
            <div
              class="index-list--cell"
            />
          </td>
          <td
            class="index-list--row-cell index-list--show-hover index-list--align-right"
          >
            <div
              class="index-list--cell"
            >
              <div
                class="confirmation-button"
              >
                <button
                  class="button button-xs button-danger"
                  tabindex="0"
                  title="Delete"
                  type="button"
                >
                  Delete
                </button>
                <div
                  class="confirmation-button--tooltip"
                >
                  <div
                    class="confirmation-button--tooltip-body"
                    data-test="confirmation-button--click-target"
                  >
                    Confirm
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            class="index-list--row-cell index-list--align-right"
          >
            <div
              class="index-list--cell"
            >
              <div
                class="context-menu"
              >
                <div
                  class="context-menu--container"
                >
                  <button
                    class="button button-xs button-primary context-menu--toggle context-menu--primary"
                    tabindex="0"
                    title="Add Data"
                    type="button"
                  >
                    Add Data
                  </button>
                  <div
                    class="context-menu--list-container"
                  >
                    <div
                      class="context-menu--list context-menu--primary"
                    >
                      <button
                        class="context-menu--item"
                      >
                        Configure Telegraf Agent
                        <div
                          class="contex-menu--item-description"
                        >
                          Configure a Telegraf agent to push data into your bucket.
                        </div>
                      </button>
                      <button
                        class="context-menu--item"
                      >
                        Line Protocol
                        <div
                          class="contex-menu--item-description"
                        >
                          Quickly load an existing line protocol file.
                        </div>
                      </button>
                      <button
                        class="context-menu--item"
                      >
                        Scrape Metrics
                        <div
                          class="contex-menu--item-description"
                        >
                          Add a scrape target to pull data into your bucket.
                        </div>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="overlay-tech"
    >
      <div
        class="overlay--dialog"
        data-test="overlay-children"
      />
      <div
        class="overlay--mask"
      />
    </div>
    <div
      class="overlay-tech"
    >
      <div
        class="overlay--dialog"
        data-test="overlay-children"
      />
      <div
        class="overlay--mask"
      />
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "unmount": [Function],
}
`;
